src/bit-fields-macros/src/lib.rs

```diff
@@ -0,0 +1,1070 @@
+                             (start..bits_len).contains(&end)` evaluated to `false` (it must \
+                             evaluate to `true`): `(0..{}).contains(&{}) && \
+                             ({}..{}).contains(&{})` => `{} && {}` => `false`",
+                            end,
+                            start,
+                            start,
+                            bits_len.size(),
+                            end,
+                            valid_start,
+                            valid_end
+                        )
+                    );
+                }
+                // Set display string
+                let more = fields_iter.peek().is_some();
+                let cropped = field_ident.to_string().chars().take(10).collect::<String>();
+                #[rustfmt::skip]
+                display_string.push_str(&format!("\
+                           ────────────{}\n\
+                    \x20  {:02}..{:02} │\n\
+                           ────────────{}\n\
+                            \x20{:>10} │\n\
+                           ────────────{}\n\
+                          \x20{{:>10}} │\n\
+                           ────────────{}\
+                    ",
+                    if more { '┬' } else { '┐' },
+                    start,
+                    end,
+                    if more { '┼' } else { '┤' },
+                    cropped,
+                    if more { '┼' } else { '┤' },
+                    if more { '┴' } else { '┘' }
+                ));
+                write!(&mut display_fmt_string, "self.{}.to_string(),", field_ident).unwrap();
+
+                // Add bit range implementations
+                let type_str = format!(
+                    "bit_fields::BitRange<{},{},{}>",
+                    struct_data_type, start, end
+                );
+                writeln!(
+                    &mut struct_member_fields,
+                    "/// {}\npub {}: {},",
+                    rustdoc, field_ident, type_str
+                )
+                .unwrap();
+                #[cfg(feature = "field_map")]
+                write!(
+                    &mut field_matching_from_hashmap,
+                    "
+                    \"{field_ident}\" => {{
+                        base.{field_ident}.checked_assign(value)?;
+                    }},
+                ",
+                    field_ident = field_ident
+                )
+                .unwrap();
+                #[cfg(feature = "field_map")]
+                write!(
+                    &mut field_setting_hashmap,
+                    "map.insert(String::from(\"{field_ident}\"),{struct_data_type}::from(&\
+                     bit_field.{field_ident}));",
+                    struct_data_type = struct_data_type,
+                    field_ident = field_ident
+                )
+                .unwrap();
+
+                writeln!(
+                    &mut struct_doc_table_layout,
```

Comment:
How is the table going to be used in cpuid?

Reply:
Used in the rustdoc documentation.